Lib/ufo2ft/featureWriters/markFeatureWriter.py

```diff
@@ -426,6 +457,62 @@ def _makeMarkToBaseAttachments(self):
             result.append(MarkToBasePos(glyphName, baseMarks))
         return result
 
+    def _makeGroupedMarkToBaseAttachments(self):
+        """Make markToBase attachements, then group them so that no group
+        contains conflicting anchor classes for the same glyph.
+        """
+        attachments = self._makeMarkToBaseAttachments()
+        # Idea: attachments is a list of mark to base pairs, linked together through an anchor name
+        # We have to put them into one or more lookups with the constraint
+        # that the same mark glyph cannot appear twice in the same lookup while
+        # using different anchor names.
+        markGlyphToMarkClasses = defaultdict(set)
+        for attachment in attachments:
+            for namedAnchor in attachment.marks:
+                for markGlyph in namedAnchor.markClass.glyphs:
+                    markGlyphToMarkClasses[markGlyph].add(namedAnchor.markClass.name)
+        # To compute the number of lookups that we need to build, we want
+        # the minimum number of lookups such that, whenever a mark glyph
+        # belongs to several mark classes, these classes are not in the same
+        # lookup. A trivial solution is to make 1 lookup per mark class
+        # but that's a bit wasteful, we might be able to do better by grouping
+        # mark classes that do not conflict.
+        # This is a graph coloring problem: the graph nodes are mark classes,
+        # edges are between classes that would conflict and the colors are
+        # the lookups in which they can go.
+        adjacency = {
+            # We'll get the same markClass several times in the dict
+            # comprehension below but it's ok, only one will be kept.
+            markClass: set()
+            for markClasses in markGlyphToMarkClasses.values()
+            for markClass in markClasses
+        }
+        for markGlyph, markClasses in markGlyphToMarkClasses.items():
+            for markClass in markClasses:
```

Comment:
you could use `itertools.combinations(markClasses, 2}`

Reply:
done